ReactNative/ReactTwitter/core/service/oauth-helper.js

```diff
@@ -0,0 +1,75 @@
+import HmacSHA1 from 'crypto-js/hmac-sha1'
+import CryptoJS from 'crypto-js'
+import PercentEncoder from './percent-encoder.js'
+
+class OauthHelper {
+  constructor (consumerSecret) {
+    this.consumerSecret = consumerSecret
+  }
+
+  buildAuthorizationHeader (method, url, params, oauthTokenSecret) {
+    let output = 'OAuth '
+    let signature = this.getSigningKey(method, url, params, oauthTokenSecret)
+    for (let key in params) {
+      output += PercentEncoder.encode(key) + '="' + PercentEncoder.encode(params[key]) + '", '
+    }
+    output += 'oauth_signature="' + PercentEncoder.encode(signature) + '"'
+    return output
+  }
+
+  getSigningKey (method, url, params, oauthTokenSecret) {
+    let signingKey = PercentEncoder.encode(this.consumerSecret) + '&' + PercentEncoder.encode(oauthTokenSecret)
+    let signatureBase = OauthHelper._getSignatureBase(method, url, params)
+    return CryptoJS.enc.Base64.stringify(HmacSHA1(signatureBase, signingKey))
+  }
+
+  static _getSignatureBase (method, url, params) {
+    return method.toUpperCase() + '&' + PercentEncoder.encode(url) + '&' + PercentEncoder.encode(OauthHelper._collectParameters(params))
+  }
+
+  static _collectParameters (params) {
+    let encodedParams = OauthHelper._percentEncodeParams(params)
+    OauthHelper._sortEncodedParams(encodedParams)
+    return OauthHelper._joinParams(encodedParams)
+  }
+
+  static _percentEncodeParams (params) {
+    let encodedParams = []
+    for (let key in params) {
+      encodedParams.push({
+        key: `${PercentEncoder.encode(key)}`,
+        value: `${PercentEncoder.encode(params[key])}`
+      })
+    }
+    return encodedParams
+  }
+
+  static _sortEncodedParams (params) {
+    params.sort((first, second) => {
+      return (first.key > second.key) ? 1 : ((second.key > first.key) ? -1 : 0)
+    })
+  }
+
+  static _joinParams (params) {
```

Comment:
also consider if it makes sense and if it makes it easier to read the code, don't just stick with it because it's a JS convention. Same way as we don't use mHungarian notation in android 🔍